### PR TITLE
Update OSS semaphore jobs that depend on glide

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,15 @@ LOCAL_GROUP_ID:=$(shell id -g)
 
 EXTRA_DOCKER_ARGS	+= -e GO111MODULE=on
 
+ifdef GOPATH
+ifneq ($(GOPATH),)
+ifndef CIRCLECI
+	LOCAL_GOPATH = $(shell echo $(GOPATH) | cut -d':' -f1)
```
Comment (ozdanborne) [inline Makefile]:
```suggestion
	# If the environment is using multiple comma-separated directories for gopath, use the first one, as that 
	# is the default one used by go modules.
	LOCAL_GOPATH = $(shell echo $(GOPATH) | cut -d':' -f1)
```

File: Makefile [hunk 2]
```diff
@@ -46,6 +46,15 @@ LOCAL_GROUP_ID:=$(shell id -g)
 
 EXTRA_DOCKER_ARGS	+= -e GO111MODULE=on
 
+ifdef GOPATH
+ifneq ($(GOPATH),)
+ifndef CIRCLECI
```
Comment (ozdanborne) [inline Makefile]:
```suggestion
# CircleCI's gopath is readonly and readonly gopaths are incompatible with go modules so don't
# volume mount the cache in that environment
ifndef CIRCLECI
```

File: Makefile [hunk 3]
```diff
@@ -46,6 +46,15 @@ LOCAL_GROUP_ID:=$(shell id -g)
 
 EXTRA_DOCKER_ARGS	+= -e GO111MODULE=on
 
+ifdef GOPATH
```
Comment (ozdanborne) [inline Makefile]:
```suggestion
# Volume-mount gopath into the build container if it's defined for persistent caching.
ifdef GOPATH
```

Comment (ozdanborne) [inline Makefile]: I think this if statement is overkill. Golang treats gopath the same whether it's set to an empty string or unset.

```
(⎈ N/A:N/A)❯ export GOPATH=""                                     (turk@tigera.io:tigera-dev)

~/tmp
(⎈ N/A:N/A)❯ go env GOPATH                                        (turk@tigera.io:tigera-dev)
/home/turk/go

~/tmp
(⎈ N/A:N/A)❯ unset GOPATH                                         (turk@tigera.io:tigera-dev)

~/tmp
(⎈ N/A:N/A)❯ go env GOPATH                                        (turk@tigera.io:tigera-dev)
/home/turk/go
```

Our makefile logic shouldn't care either.

Can we just rely on the following `ifneq` statement?

Comment (rafaelvanoni) [reply Makefile]: I think so

**Pull request merged**